src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerHandler.cs

```diff
@@ -75,38 +74,50 @@ public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRe
 
         public override Task Save(ServerViewContext context, ServerInfo obj)
         {
-            UpdateServerProperties(context.Parameters, obj);
+            UpdateServerProperties(context.Parameters, obj, RunType.RunNow);
             return Task.CompletedTask;
         }
 
         public override Task<string> Script(ServerViewContext context, ServerInfo obj)
         {
-            throw new NotSupportedException("ServerHandler does not support Script method");
+            var script = UpdateServerProperties(
+                 context.Parameters,
+                 obj,
+                 RunType.ScriptToWindow);
+            return Task.FromResult(script);
         }
 
-        private void UpdateServerProperties(InitializeViewRequestParams viewParams, ServerInfo serverInfo)
+        private string UpdateServerProperties(InitializeViewRequestParams viewParams, ServerInfo serverInfo, RunType runType)
         {
-            if (viewParams != null)
+            if (viewParams == null)
             {
-                ConnectionInfo connInfo = this.GetConnectionInfo(viewParams.ConnectionUri);
-                CDataContainer dataContainer = CDataContainer.CreateDataContainer(connInfo);
-
-                ServerPrototype prototype = new ServerPrototype(dataContainer);
-                prototype.ApplyInfoToPrototype(serverInfo);
-                ConfigureServer(dataContainer, ConfigAction.Update, RunType.RunNow, prototype);
+                
             }
+            ConnectionInfo connInfo = this.GetConnectionInfo(viewParams.ConnectionUri);
+            CDataContainer dataContainer = CDataContainer.CreateDataContainer(connInfo);
+
+            ServerPrototype prototype = new ServerPrototype(dataContainer);
+            prototype.ApplyInfoToPrototype(serverInfo);
+            return ConfigureServer(dataContainer, ConfigAction.Update, runType, prototype);
```

Comment:
Make sure to disconnect the connection for the dataContainer in a finally statement so we aren't holding connections open. So `dataContainer.ServerConnection.Disconnect()`